src/main/java/ch/njol/skript/expressions/ExprRandomNumber.java

```diff
@@ -33,68 +35,71 @@
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
-import ch.njol.util.Math2;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Random Number")
-@Description({"A random number or integer between two given numbers. Use 'number' if you want any number with decimal parts, or use use 'integer' if you only want whole numbers.",
-		"Please note that the order of the numbers doesn't matter, i.e. <code>random number between 2 and 1</code> will work as well as <code>random number between 1 and 2</code>."})
-@Examples({"set the player's health to a random number between 5 and 10",
-		"send \"You rolled a %random integer from 1 to 6%!\" to the player"})
+@Description({
+	"A random number or integer between two given numbers. Use 'number' if you want any number with decimal parts, or use use 'integer' if you only want whole numbers.",
+	"Please note that the order of the numbers doesn't matter, i.e. <code>random number between 2 and 1</code> will work as well as <code>random number between 1 and 2</code>."
+})
+@Examples({
+	"set the player's health to a random number between 5 and 10",
+	"send \"You rolled a %random integer from 1 to 6%!\" to the player"
+})
 @Since("1.4")
 public class ExprRandomNumber extends SimpleExpression<Number> {
+
 	static {
 		Skript.registerExpression(ExprRandomNumber.class, Number.class, ExpressionType.COMBINED,
-				"[a] random (1¦integer|2¦number) (from|between) %number% (to|and) %number%");
+				"[a] random (:integer|number) (from|between) %number% (to|and) %number%");
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<? extends Number> lower, upper;
-	
-	private final Random rand = new Random();
-	
+
+	private Expression<? extends Number> from, to;
+
+	private final Random random = ThreadLocalRandom.current();
+
 	private boolean integer;
```

Comment:
Would be better to change `integer` to `isInteger`
```suggestion
	private boolean isInteger;
	private Expression<? extends Number> from, to;
	private final Random random = ThreadLocalRandom.current();
```